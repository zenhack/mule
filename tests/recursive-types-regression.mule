#XFAIL: type formatting
#
# Regression test for a stack overflow originally fixed by commit
# b28750051945886aa838582660c4cd167951d037.
#
(fn x. x x) (fn x. x)
