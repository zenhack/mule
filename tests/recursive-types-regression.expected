
Desugared:

	((fn x (x x)) (fn x x))

inferred type:

	(all a (a -> a))

Runtime term:

	(App ((Lam (0 () (App ((Var 0) (Var 0))))) (Lam (0 () (Var 0)))))

Evaluated:

	(Lam (0 () (Var 0)))
